lib/tests/test.js

```diff
@@ -41,18 +41,27 @@ var Test = function(options) {
   });
 
   this.web3 = new Web3();
-};
-
-Test.prototype.deployAll = function(contractsConfig, cb) {
-  var self = this;
-
+  
   if (this.simOptions.node) {
     this.web3.setProvider(new this.web3.providers.HttpProvider(this.simOptions.node));
   } else {
     this.sim = getSimulator();
     this.web3.setProvider(this.sim.provider(this.simOptions));
   }
+};
 
+Test.prototype.getAccounts = function(cb) {
+  this.web3.eth.getAccounts(function(err, accounts) {
+    if (err) {
+      throw new Error(err);
+    }
+    cb(accounts);
```

Comment:
Instead of throwing on error, you should callback the error and when it is a success: `callback(null, accounts)`. That's what web3 does for example. Also, basically the standard when using callbacks (error is always the first parameter)

Reply:
@jrainville I copied the code from here https://github.com/embark-framework/embark/blob/develop/lib/tests/test.js#L107

should it be changed there too?

Reply:
@jrainville did the PR for your change here https://github.com/embark-framework/embark/pull/442